pennylane/ops/qubit.py

```diff
@@ -565,6 +567,96 @@ def decomposition(phi, theta, omega, wires):
         return decomp_ops
 
 
+class PauliRot(Operation):
+    r"""PauliRot(theta, pauli_word, wires)
+    Arbitrary pauli word rotation.
+
+    .. math::
+
+        RP(\theta, P) = \exp(-i \frac{\theta}{2} P)
+
+    **Details:**
+
+    * Number of wires: Any
+    * Number of parameters: 2 (1 differentiable parameter)
+    * Gradient recipe: :math:`\frac{d}{d\theta}f(RP(\theta)) = \frac{1}{2}\left[f(RP(\theta +\pi/2)) - f(RP(\theta-\pi/2))\right]`
+      where :math:`f` is an expectation value depending on :math:`RP(\theta)`.
+
+    .. note::
+
+        If the ``PauliRot`` gate is not supported on the targeted device, PennyLane
+        will attempt to decompose the gate using :class:`~.RX`, :class:`~.Hadamard`, :class:`~.RZ`
+        and :class:`~.CNOT` gates.
+
+    Args:
+        theta (float): rotation angle :math:`\theta`
+        pauli_word (string): the Pauli word defining the rotation
+        wires (Sequence[int] or int): the wire the operation acts on
+    """
+    num_params = 2
+    num_wires = Any
+    do_check_domain = False
+    par_domain = "R"
+    grad_method = "A"
+
+    _PAULI_CONJUGATION_MATRICES = {
+        "X": Hadamard._matrix(),
+        "Y": RX._matrix(np.pi / 2),
+        "Z": np.eye(2),
+    }
+
+    @staticmethod
+    def _matrix(*params):
+        theta = params[0]
+        pauli_word = params[1]
+        # We keep imaginary wires to be able to reconstruct the position of the identities
+        active_wires, active_gates = zip(
+            *[(wire, gate) for wire, gate in enumerate(pauli_word) if gate != "I"]
+        )
+
+        # we first create the multi-Z rotation
+        multi_Z_rot_eigs = np.exp(-1j * theta / 2 * pauli_eigs(len(active_gates)))
+        multi_Z_rot_matrix = np.diag(multi_Z_rot_eigs)
+
+        # now we conjugate with Hadamard and RX to create the Pauli string
+        conjugation_matrix = functools.reduce(
+            np.kron, [PauliRot._PAULI_CONJUGATION_MATRICES[gate] for gate in active_gates],
+        )
```

Comment:
From a maintainability point of view, I'm wondering if it was worth building a `Tensor` here. The reason being is that this logic is contained there as well in `_matrix`. Therefore it might be cool to use that, as then we will only need to take care of making that part.

Reply:
I'd expect that the creation of the `Tensor` should not result in significant overhead (the `__init__` seems to be pretty  light)

Reply:
I was refraining from this for two reasons. First, you have the overhead from both the `Tensor` and `Observable` instances. Second, technically, `Tensor` is a tensor of observables and not of unitaries. The Paulis just happen to be both. Also, creating the `Tensor` would not be less complicated, I guess something along the lines of
```python
qml.Tensor(*[_PAULIS[gate](wire) for wire, gate in zip(active_wires, active_gates)]).matrix
```
So I'd prefer to leave it as is

Reply:
I think you're right, that's a fair point! We currently only ever use `Tensor` for observables, so it might confuse the logic to have it in this context. The cleanest would be to have a "unitary" equivalent of `Tensor`, though let's see if there would be use cases for that (definitely out of scope here). Happy if this stays as is! :)